lvsfunc/kernels.py

```diff
@@ -0,0 +1,95 @@
+"""
+Kernels for vapoursynth internal resizers. Intended for use by
+:py:mod:`lvsfunc.scale` functions.
+"""
+from abc import ABC, abstractmethod
+
+import vapoursynth as vs
+
+core = vs.core
+
+
+class Kernel(ABC):
+    """ Abstract scaling kernel interface. """
+    @abstractmethod
+    def scale(self, clip: vs.VideoNode, width: int, height: int, **kwargs) -> vs.VideoNode:
```

Comment:
What kind of kwargs would the library put here? As far as I can tell, anything it would want to use should be properly communicated in this signature, which from looking at the code below would be `src_left` and `src_top`. In that case, those should be added to the signature with default values and `**kwargs` be removed from the calls.
```py
def scale(self, clip: vs.VideoNode, width: int, height: int, *, src_left: float, src_top: float) -> vs.VideoNode:
```
And they should probably receive default values in the subclasses.
Unless this isn't flexible enough and you can forsee many other parameters potentially being used that are simultaneously supported by all kernels.

Now, to allow more customization of the kernel, a `**kwargs` could be added to the constructur (of the abstract class or implementations) that is used in the calls like `core.resize.Bilinear(clip, width, height, **self.kwargs)`.